apps/api/src/server/applications/subscriptions/subscriptions.controller.js

```diff
@@ -0,0 +1,144 @@
+import * as subscriptionRepository from '../../repositories/subscription.respository.js';
+import { eventClient } from '../../infrastructure/event-client.js';
+import { EventType } from '@pins/event-client';
+import { NSIP_SUBSCRIPTION } from '../../infrastructure/topics.js';
+import { buildSubscriptionPayload } from './subscriptions.js';
+import logger from '../../utils/logger.js';
+import { Prisma } from '@prisma/client';
+
+/**
+ * @type {import('express').RequestHandler}
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+export async function getSubscription(request, response) {
+	const { query } = request;
+
+	const caseReference = String(query.caseReference);
+	const emailAddress = String(query.emailAddress);
+
+	try {
+		const res = await subscriptionRepository.findUnique(caseReference, emailAddress);
+
+		if (res === null) {
+			response.status(404).send({ errors: { notFound: 'subscription not found' } });
+		} else {
+			response.send(res);
+		}
+	} catch (/** @type {any} */ e) {
+		/** @type {Object<string, string>} */
+		const errors = {};
+		logger.warn(e, 'unhandled error');
+		errors.unknown = e.message || 'unknown';
+		response.status(400).send({ errors });
+	}
+}
+
+/**
+ * @type {import('express').RequestHandler}
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+export async function createSubscription(request, response) {
+	const { body } = request;
+
+	/** @type {import('@prisma/client').Prisma.SubscriptionCreateInput} */
+	const subscription = {
+		caseReference: body.caseReference,
+		emailAddress: body.emailAddress,
+		subscriptionType: body.subscriptionType
+	};
+
+	if (body.startDate) {
+		subscription.startDate = new Date(body.startDate);
+	}
+	if (body.endDate) {
+		subscription.endDate = new Date(body.endDate);
+	}
+	if (body.language) {
+		subscription.language = body.language;
+	}
+
+	try {
+		const res = await subscriptionRepository.create(subscription);
```

Comment:
Since we're doing this soft deletion with subscriptions, what happens if somebody subscribes, unsubscribes and then subscribes again?

Reply:
I think we update the `PATCH` endpoint to allow setting both `startDate` and `endDate`, and in the function, if we get a `Create` event for a subscription that already exists, we update `startDate=now` and `endDate=null` accordingly. Any thoughts?

Reply:
Yeah I think turning this into an upsert operation makes most sense to me!

Reply:
Have done this change: https://github.com/Planning-Inspectorate/back-office/pull/1029/commits/943468f90c51f26f77517f1f506f953daa26d1cf